### PR TITLE
feat: persist schema generator examples

### LINKED ISSUE #346: [Feature] Persist edited schema generator query and few-shot examples in the demo
### Search before asking

- [x] I had searched in the [feature](https://github.com/apache/hugegraph-ai/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

## Feature Description

The graph schema generator lets users edit query examples and few-shot schema
examples, but those edits are not persisted. Main document text, graph schema,
and graph extraction prompt header are already persisted through the prompt YAML
flow. Please extend persistence to schema-generator examples.

## Current verification

- `store_prompt(doc, schema, example_prompt)` only saves `doc_input_text`, `graph_schema`, and `extract_graph_prompt` in `hugegraph-llm/src/hugegraph_llm/demo/rag_demo/vector_graph_block.py:47`.
- The schema generator UI has editable `query_example` and `few_shot` code blocks in `hugegraph-llm/src/hugegraph_llm/demo/rag_demo/vector_graph_block.py:284` and `:291`.
- The build-schema click chain has an explicit TODO: `Store the updated examples` in `hugegraph-llm/src/hugegraph_llm/demo/rag_demo/vector_graph_block.py:358`.
- `BuildSchemaFlow` already receives `query_examples` and `few_shot_schema`; the missing part is persistence/reload behavior, not schema generation input wiring.

## Suggested scope

- Add config fields for schema-generator query examples and few-shot examples, or add a separate persisted demo state file if that better matches existing config conventions.
- Load persisted examples into the UI before falling back to bundled resource examples.
- Persist edited examples when users generate schema or perform related save-triggering actions.
- Validate that edited examples are valid JSON before saving.
- Keep bundled examples read-only as defaults; do not overwrite files under `resources/prompt_examples`.

## Mermaid reference

```mermaid
flowchart TD
    UI[Schema generator code blocks] --> Validate[Validate JSON]
    Validate -->|valid| Save[Persist user examples]
    Validate -->|invalid| Error[Show JSON error]
    Save --> Reload[Demo reload]
    Defaults[Bundled resource examples] --> Load{User saved examples?}
    Save --> Load
    Load -->|yes| UI
    Load -->|no| Defaults
```

## Acceptance criteria

- Edited query examples survive demo refresh/restart.
- Edited few-shot schema examples survive demo refresh/restart.
- Invalid JSON is not persisted and produces a clear UI error.
- Default bundled examples still load for first-time users.

## Suggested tests

- Unit test for save/load helper behavior.
- UI helper test showing edited values are passed to the persistence function.
- Config test to ensure old prompt config files without these fields still load.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by nw9663644-eng**
## Purpose

Closes #346.

This PR persists edited schema-generator query examples and few-shot schema examples in the demo.

## Changes

* Added prompt config fields for schema-generator query examples and few-shot schema examples.
* Load persisted schema-generator examples into the demo UI before falling back to bundled resource examples.
* Persist edited examples through the existing prompt config save path.
* Validate edited examples as JSON before saving.
* Reject invalid JSON with a clear UI error and avoid persisting invalid content.
* Keep bundled examples under `resources/prompt_examples` read-only as defaults.
* Added tests for save/load behavior, invalid JSON handling, bundled fallback behavior, invalid persisted fallback behavior, UI persistence wiring, and old prompt config compatibility.

## Tests

* `uv run ruff check hugegraph-llm/src/hugegraph_llm/config/models/base_prompt_config.py hugegraph-llm/src/hugegraph_llm/demo/rag_demo/vector_graph_block.py hugegraph-llm/src/tests/document/test_schema_generator_examples_persistence.py`
* `uv run ruff format --check .`
* `uv run pytest hugegraph-llm/src/tests/document/test_schema_generator_examples_persistence.py`
* `uv run pytest hugegraph-llm/src/tests/document`

### CODE REVIEW

File: hugegraph-llm/src/hugegraph_llm/demo/rag_demo/vector_graph_block.py [hunk 1]
```diff
@@ -44,13 +44,59 @@
...
+
+def store_prompt(
+    doc,
+    schema,
+    example_prompt,
+    graph_extract_split_type="document",
+    query_examples=None,
+    few_shot_examples=None,
+):
+    validated_query_examples = _validate_schema_generator_examples(query_examples, "Query examples")
```
Comment (imbajin) [inline hugegraph-llm/src/hugegraph_llm/demo/rag_demo/vector_graph_block.py]: ⚠️ **Validate schema examples before generic actions**

Evidence: these save chains now run the primary action first and only then call `store_prompt(..., query_example, few_shot)`. Since `store_prompt()` validates those JSON blobs immediately, an invalid or blank schema-example editor can make actions like clearing indexes report an error after the action already ran, and blank values cannot clear a persisted override back to bundled defaults.

Please keep schema-generator validation/persistence on an explicit schema-generator save/generate path, or validate before the primary action and treat blank/whitespace examples as clearing the persisted override. Add a regression test for clearing both fields back to the bundled fallback.

**Review by imbajin — COMMENTED** [review]
Blocking: no. Summary: Schema-example persistence currently validates after unrelated UI actions, which can make successful actions look failed. Evidence: static diff in vector_graph_block.py, targeted pytest passed after uv sync, and latest-head CI has no visible failures.

File: hugegraph-llm/src/hugegraph_llm/demo/rag_demo/vector_graph_block.py [hunk 2]
```diff
@@ -44,7 +44,56 @@
 )
 
 
-def store_prompt(doc, schema, example_prompt, graph_extract_split_type="document"):
+def _validate_schema_generator_examples(examples, label):
+    examples = (examples or "").strip()
+    if not examples:
+        return ""
+    try:
+        json.loads(examples)
```
Comment (imbajin) [inline hugegraph-llm/src/hugegraph_llm/demo/rag_demo/vector_graph_block.py]: ‼️ **Validate the schema-example shape, not only JSON syntax**

Evidence: the persisted/bundled `query_examples.json` is a `list[str]`, but `SchemaBuildNode` only keeps parsed items that are dicts containing `description` and `gremlin`; the current head therefore reloads and persists examples that are later collapsed to an empty list before schema generation. The new tests cover YAML persistence and JSON syntax, but not that persisted query examples actually reach the schema prompt.

Impact: users can save examples successfully and still generate schemas without any query examples, so this feature appears to work while silently dropping the main payload. Please align this validator with the downstream contract, or relax the downstream parser to accept the shipped string-list format, and add a regression test that persisted/bundled query examples produce a non-empty `query_examples` payload for schema generation.

**Review by imbajin — COMMENTED** [review]
Blocking: yes. Summary: Found one schema-example contract regression in the current head. Evidence: target pytest passed after uv sync; static check shows persisted bundled query examples are filtered out before schema generation.

File: hugegraph-llm/src/hugegraph_llm/demo/rag_demo/vector_graph_block.py [hunk 3]
```diff
@@ -44,7 +44,56 @@
...
+    except json.JSONDecodeError as exc:
+        log.warning("Ignoring invalid persisted %s: %s", label, exc)
+        return ""
+    return examples
+
+
+def _persist_schema_generator_examples(query_examples, few_shot_examples):
+    validated_query_examples = _validate_schema_generator_examples(query_examples, "Query examples")
+    validated_few_shot_examples = _validate_schema_generator_examples(few_shot_examples, "Few-shot schema examples")
```
Comment (VGalaxies) [inline hugegraph-llm/src/hugegraph_llm/demo/rag_demo/vector_graph_block.py]: **Medium: Validate few-shot schema shape before persisting**

`hugegraph-llm/src/hugegraph_llm/demo/rag_demo/vector_graph_block.py:72`

**Evidence**
- `_validate_schema_generator_examples()` only checks `json.loads()`, so `_persist_schema_generator_examples()` accepts and saves values like `[{"schema": {"vertices": []}}]`; the new tests use that array shape. Downstream, `SchemaBuildNode` passes the parsed value through as `few_shot_schema`, and `SchemaBuilder.run()` rejects anything that is not a dict with `"'few_shot_schema' must be a dict"`.

**Impact**
- A user can save syntactically valid few-shot examples that make schema generation fail, then the invalid persisted override keeps reloading until manually cleared.

**Requested fix**
- Validate `few_shot_examples` as the object/dict shape required by schema generation before writing it to YAML, or update the downstream flow to intentionally accept the persisted shape and cover that path with a regression test.

**Review by VGalaxies — COMMENTED** [review]
**Review summary**

- **Blocking:** yes
- **Summary:** The PR can persist few-shot schema examples that pass JSON syntax validation but are rejected by the schema-generation flow.
- **Evidence:**
  - `git diff --check origin/main...HEAD`
  - static inspection of the changed validator/persistence path and downstream schema builder contract

**Review by imbajin — APPROVED** [review]
Blocking: no. Summary: No obvious issues found in the current head. Evidence: static review of schema-example normalization/persistence plus latest-head CI success; local targeted pytest could not run because nltk is missing in the temporary uv environment.

**Review by imbajin — APPROVED** [review]
